asab/pdict.py

```diff
@@ -3,71 +3,50 @@
 import shelve
 
 
-class PersistentDict(collections.MutableMapping):
+class PersistentDict(dict):
 
 	"""
 	The persistent dictionary works as the regular Python dictionary but the content of the dictionary is stored in the file.
 	You cat think of a ``PersistentDict`` as a simple `key-value store <https://en.wikipedia.org/wiki/Key-value_database>`_.
 	It is not optimized for a frequent access. This class provides common ``dict`` interface.
 
+	*Warning*: You must explicitly `load()` and `store()` content of the dictionary
 	*Warning*: You can only store objects in the persistent dictionary that are serializable.
 	"""
 
 	def __init__(self, path):
+		super().__init__()
 		# Create directory, if needed
 		dirname = os.path.dirname(path)
 		if not os.path.isdir(dirname):
 			os.makedirs(dirname)
 
 		self._path = path
 
-	def __getitem__(self, key):
-		with shelve.open(self._path) as d:
-			return d[key]
-
-	def __setitem__(self, key, value):
-		with shelve.open(self._path) as d:
-			d[key] = value
-
 	def __delitem__(self, key):
+		super().__delitem__(key)
 		with shelve.open(self._path) as d:
 			del d[key]
 
-	def __iter__(self):
-		with shelve.open(self._path) as d:
-			for key in d:
-				yield key
-
-	def __len__(self):
-		with shelve.open(self._path) as d:
-			return len(d)
-
-	def __str__(self):
+	def load(self) -> None:
+		"""
+		Load content of file as dictionary.
+		"""
 		with shelve.open(self._path) as d:
-			return str(dict(d))
+			for key, value in d.items():
+				self[key] = value
 
-	def load(self, *keys):
+	def store(self) -> None:
 		"""
-		D.load([keys]) -> [values].
-
-		Optimised version of the get() operations that load multiple keys from the persistent store at once.
-		It saves IO in exchange for possible race conditions.
-
-		:param keys: A list of keys.
-		:return: A list of values in the same order to provided key list.
-
-		.. code:: python
-
-			v1, v2, v3 = pdict.load('k1', 'k2', 'k3')
-
+		Explicitly store content of persistent dictionary to file
 		"""
+
 		with shelve.open(self._path) as d:
-			return (d[key] for key in keys)
+			for key, value in self.items():
+				d[key] = value
```

Comment:
@SukiCZ Does it automatically save also nested dictionaries/arrays?

Reply:
Yup. I extended the example (didn't commit it tho) and tried nested dicts which was working fine.